src/base/static/components/input-form/input-form.js

```diff
@@ -0,0 +1,776 @@
+    // The pre-save hook allows flavors to attach arbitrary data to the attrs
+    // object before submission to the database.
+    if (customHooks && customHooks.preSave) {
+      attrs = hooks[customHooks.preSave](attrs);
+    }
+
+    model.save(attrs, {
+      success: (response) => {
+        Util.log("USER", "new-place", "successfully-add-place");
+        this.setState({ "formIsSubmitting": false });
+
+        // Fire post-save hook.
+        // The post-save hook allows flavors to hijack the default
+        // route-to-detail-view behavior.
+        if (customHooks && customHooks.postSave) {
+          customHooks.postSave(response, model, this.defaultPostSave.bind(this));
+        } else {
+          this.defaultPostSave(model);
+        }
+      },
+      error: (error) => {
+        Util.log("USER", "new-place", "fail-to-add-place");
+      },
+      wait: true,
+    });
+  }
+
+  defaultPostSave(model) {
+    this.reset();
+    this.props.router.navigate(Util.getUrl(model), { trigger: true });
+  }
+
+  getVisibleFormCategories() {
+    return this.props.placeConfig.place_detail
+      // Filter out categories that shouldn't display on the form at all.
+      .filter(categoryConfig => categoryConfig.includeOnForm)
+      // Filter out admin_only categories that shouldn't be shown given the
+      // current user's credentials.
+      .filter(categoryConfig => {
+        if (categoryConfig.admin_only &&
+            !Util.getAdminStatus(categoryConfig.dataset, categoryConfig.admin_groups)) {
+          return false;
+        }
+        return true;
+      });
+  }
+
+  fieldIsInvalid(fieldName) {
+    return (
+      this.state.formValidationErrors.length > 0 
+      && this.requiredFields.includes(fieldName)
+      && !this.state.fieldValues[fieldName]
+    );
+  }
+
+  render() {
+    const { categoryMenuIsCollapsed, categoryMenuIsHidden, formIsSubmitting,
+            formValidationErrors, selectedCategory } = this.state;
+    const { className, hideCenterPoint, hideSpotlightMask, placeConfig, 
+            showNewPin } = this.props;
+    const classNames = {
+      categoryBtns: cn("input-form__category-buttons-container", {
+        "input-form__category-buttons-container--visible": !categoryMenuIsHidden,
+        "input-form__category-buttons-container--hidden": categoryMenuIsHidden
+      }),
+      form: cn("input-form__form", className, {
+        "input-form__form--active": !formIsSubmitting,
+        "input-form__form--inactive": formIsSubmitting
+      }),
+      warningMsgs: cn("input-form__warning-msgs-container", {
+        "input-form__warning-msgs-container--visible": formValidationErrors.length > 0,
+        "input-form__warning-msgs-container--hidden": formValidationErrors.length === 0
+      }),
+      spinner: cn("input-form__submit-spinner", {
+        "input-form__submit-spinner--visible": formIsSubmitting,
+        "input-form__submit-spinner--hidden": !formIsSubmitting
+      })
+    };
```

Comment:
This is the same issue that I mentioned in the other input-form component - can't we use a default value for when the form is visible, and set the `--hidden` modifier` only?

Reply:
Yep, I'll update this throughout the PR :)